### ISSUE TITLE
[FE] My VA, Onsite Notifications: Add Alt Text for the icons on all alerts and notifications on My VA

### ISSUE THREAD

**Issue by hdjustice** [department-of-veterans-affairs/va.gov-team#36216]
## Background
From the Midpoint Review for Onsite Notifications, we received Accessibility feedback regarding the icons within the On-site notification message box and should consider adding Alt Text for each of them.

See Mid-point Review ticket #34621 for Onsite notifications, **[Josh's feedback](https://github.com/department-of-veterans-affairs/va.gov-team/issues/34621#issuecomment-1016708550)**

See also Staging Review ticket #38376 for Payment Info.  Tickets listed below are findings for us to address Alt Text on _**all alerts that display on My VA.**_
- #39473
- #39474 

**NOTE** - The preceding icon may change, will need update this ticket with any related information before we start working on it.

## Tasks

- [x] Confirm language with content for alt text (likely should be "important", but Angela to get some guidance/confirmation)
- [ ] Add Alt Text to the preceding icon that identifies the type of alert or notification 
     - "Alt text for the "!" icon in dismissible messages (e.g. should it be "Important" vs. "Alert"-- might need some content advice here). I'd recommend thinking ahead of time how these might be reworded based on how important they are"
- [ ] Add Alt Text to the dismissible 'X' icon 
     - "Alt text for the "X" icon in dismissible messages"

## Acceptance Criteria
- Alt Text is present to identify the type of alert icon and dismissible 'X' icon

**Comment by hdjustice** [comment]
@hdjustice - Double check on this one. May need to be addressed by Design system team (not our team individually)
Dismiss X has aria-label

**Comment by hdjustice** [comment]
Feedback via slack discussion in design channel

Josh
>  I think that for the time being teams would be responsible for creating it for each icon used as it isn't in the design system yet (example: "Alert" describing its purpose/meaning, not its visual shape).

Brooks
> We don't currently have a system-wide icon strategy for handling icons and accessibility (but we have a [ticket](https://github.com/department-of-veterans-affairs/va.gov-team/issues/32758) to explore that). Most of the icons are <i> tags using Font Awesome glyphs, and a few others are <svg> elements. When we use those internally on the components that the design system provides we'll set the aria-label for icons not hidden from the screen reader. For app teams using Font Awesome icons it will be up to them to set aria-label as appropriate.
There is a CI job that will [leave comments on PRs where it sees an ](https://github.com/department-of-veterans-affairs/vets-website/blob/1a02e023e390ff4a3748bf477611b2206839fdb1/.github/workflows/pull-request.yml#L126-L141)<i>[ tag used](https://github.com/department-of-veterans-affairs/vets-website/blob/1a02e023e390ff4a3748bf477611b2206839fdb1/.github/workflows/pull-request.yml#L126-L141), but I don't think there's anything else at the moment reminding teams to make their icons accessible.
I believe it's more common for icons on our components to be hidden from screen readers, but if their presence is semantic then it will be communicated elsewhere. [Here's an example](https://github.com/department-of-veterans-affairs/component-library/blob/e157e3c011afb64c8431cc7c7b266ab4c786ac03/packages/web-components/src/components/va-alert/va-alert.tsx#L171-L177) - the icon has aria-hidden="true" and is inside of a button which has "Close notification" as the default aria-label.

**Comment by hdjustice** [comment]
Need to find out how many alerts are present on My VA that would need the Alt Text set up for the icons.

**Comment by aagosto90** [comment]
Since there is not guidance in the design system for these icons' alt text and Josh said it's up to individual teams to decide for now, I'm going to recommend "important" for our notifications and "warning" for all other alerts if that's possible. I would recommend this because it's technically a warning alert and I think warning works for a system error which is what these alert styles are supposed to be for, but for our notifications MVP "warning" doesn't make as much sense. @tmitchellgcio

**Comment by tmitchellgcio** [comment]
@aagosto90 After discussion with @joshkimux I am going to use `sr-only` tags and the dismissible icon already has a label for screen readers on it

**Comment by ajakabcin** [comment]
Closing this ticket as it should be handled by design system. Angela submitted the design system ticket as noted above.
